File: web-forms/src/store/evaluation/contato.js
import { defineStore } from 'pinia'
import { ref } from 'vue';

const responsavelBase = {
    nome: null,
    parentesco: null,
    email: null,
    telefone: null,
    forma_contato: null,
    numero_prioritario: false
}

export const contatoModel = {
    telefone_residencial: null,
    telefone_paciente: null,
    email_paciente: null,

    responsaveis: [{ ...responsavelBase }]
}

export const useContatoStore = defineStore('contato', () => {

    const contato = ref(JSON.parse(JSON.stringify(contatoModel)))

    function adicionarResponsavel() {
        contato.value.responsaveis.push({ ...responsavelBase })
    }

    function removerResponsavel(index) {
        if (contato.value.responsaveis.length > 1) {
            contato.value.responsaveis.splice(index, 1)
        }
    }

    return {
        contato,
        adicionarResponsavel,
        removerResponsavel
    }
}, {
    persist: true
})

File: backend/internal/avaliation/domain/contato.go
package domain

type Contato struct {
	TelefoneResidencial string        `json:"telefone_residencial"`
	TelefonePaciente    string        `json:"telefone_paciente"`
	EmailPaciente       string        `json:"email_paciente"`
	Responsaveis        []Responsavel `json:"responsaveis"`
}

type Responsavel struct {
	Nome              string `json:"nome"`
	Parentesco        string `json:"parentesco"`
	Email             string `json:"email"`
	Telefone          string `json:"telefone"`
	FormaContato      string `json:"forma_contato"`
	NumeroPrioritario bool   `json:"numero_prioritario"`
}
